### PR TITLE
feat: migrate get encrypt metadata endpoint controller to TypeScript

### PR DESCRIPTION
**Pull request by karrui**
## Problem

Building upon #601, this PR migrates the rest of the controller into Typescript.

## Solution

**Features**:

- add `EncryptSubSvc#getSubmissionMetadata` service fn to use model static method
- add `EncryptSubSvc#getSubmissionMetadataList` service fn to use model static method
- add `EncryptSubCtl#handleGetMetadata` handler fn for route endpoint

**Improvements**:

- Tighten (or relax depending on your POV) Joi validation on metadata retrieval endpoint
  - `query.page` is now optional iff `query.submissionId` exists, since singular submission metadata search does not require the page parameter.

## Tests

Add tests for all new methods. 
Note that route integration tests are not done in this PR, and will be done when route also gets refactored.

### CODE REVIEW

**Review by tshuli — APPROVED** [review]
lgtm

**Review by mantariksh — APPROVED** [review]
1 question

File: src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts [hunk 1]
```diff
@@ -205,3 +209,53 @@ export const transformAttachmentMetasToSignedUrls = (
...
   )
 }
+
+export const getSubmissionMetadata = (
+  formId: string,
+  submissionId: string,
+): ResultAsync<SubmissionMetadata | null, DatabaseError> => {
+  // Early return, do not even retrieve from database.
+  if (!mongoose.Types.ObjectId.isValid(submissionId)) {
+    return okAsync(null)
```
[the remark this answers is not in the document]
Comment (karrui) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts:219]: this is due to frontend sending possible invalid submissionIds due to search, and not handling if we return 400 an error is shown to the user resulting in confusion.

returning nothing means no result found, but it's still a valid search term i guess.

We can change it if you want, just not in this PR.

Related issue opened for the debounce thingy: #603

[the remark this answers is not in the document]
Comment (mantariksh) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts:219]: ok sounds good

**Pull request merged**